exercise/cache/points.py

```diff
@@ -22,28 +29,38 @@
 from .hierarchy import ContentMixin
 
 
-def has_more_points(submission: Submission, current_entry: Dict[str, Any]) -> bool:
-    if submission.grade == current_entry['points']:
-        return is_newer(submission, current_entry)
-    return submission.grade >= current_entry['points']
+# Type for the submission compare functions.
+IsBetterThan = Callable[[Submission, Optional[Submission]], bool]
```

Comment:
Sometimes I have seen `if TYPE_CHECKING:` before listing static types. I'm just curious if this line would be a typical place to use the `if` or not.

(I am still not very familiar with static typing.)

Reply:
Could be done either way. The reason `if TYPE_CHECKING` is usually used is when we want to do things that have no runtime effect. For example:
- For type hinting purposes, importing classes that only exist in the `django-types` type library but not in Django. (`RelatedManager`)
- For type hinting purposes, importing classes that would cause a cyclical import error at runtime.
- Type hinting model fields that Django adds at runtime (id, reverse relationships).

There isn't really a reason to use it here, but it could still be done because it kind of makes sense - `IsBetterThan` has no runtime purpose, it only exists so I could add a type to the `is_better_than` variable, which was initialized to `None`.

Reply:
I actually decided to remove this `IsBetterThan` type and other initial types from the submission loop variables because they can be inferred by the IDE, and it was technically wrong to initialize them to `None` while annotating them with another type: https://github.com/apluslms/a-plus/pull/911/commits/723b6f8d18be7dea07f4599cb96a7a68ccfde27a.

Reply:
Sounds good!